README.md

```diff
@@ -89,19 +94,96 @@ int main()
+#### Key Core concepts
+
+Understanding the key concepts from the `azure-core-cpp` library, which is leveraged by all client libraries will also be helpful in getting started, regardless of which Azure service you want to use.
+
+The main shared concepts of `Azure::Core` include:
+
+- Accessing HTTP response details for the returned model of any SDK client operation, via `Response<T>`.
+- Exceptions for reporting errors from service requests in a consistent fashion via the base exception type `RequestFailedException`.
+- Abstractions for Azure SDK credentials (`TokenCredential`).
+- Handling streaming data and input/output (I/O) via `BodyStream` along with its derived types.
+- Polling long-running operations (LROs), via `Operation<T>`.
+- HTTP pipeline and HTTP policies such as retry and logging, which are configurable via service client specific options.
+- Replaceable HTTP transport layer to send requests and receive responses over the network.
+
+#### `Response <T>` Model Types
+
+Many client library operations **return** the templated `Azure::Core::Response<T>` type from the API calls. This type let's you get the raw HTTP response from the service request call the Azure service APIs make, along with the result of the operation to get more API specific details. This is the templated `T` operation result which can be extracted from the response, using the `Value` field.
+
+```C++
+  std::string containerName = "testcontainer";
+
+  BlobContainerClient containerClient
+      = BlobContainerClient::CreateFromConnectionString(GetConnectionString(), containerName);
+
+  // Azure service operations return a Response<T> templated type
+  Azure::Response<Models::CreateBlobContainerResult> createResult
+      = containerClient.CreateIfNotExists();
+
+  // You can extract the T, from the returned Response<T>,
+  // which is typically named with a Result suffix in the type name.
+  Models::CreateBlobContainerResult result = createResult.Value;
+
+  // Now you can look at API specific members on the result object that is returned.
+  printf("Did the container get created? %d\n", result.Created);
+```
+
+#### Long Running Operations
+
+Some operations take a long time to complete and require polling for their status. Methods starting long-running operations return `Operation<T>` types.
+
+You can intermittently poll whether the operation has finished by using the `Poll()` method on the returned `Operation<T>` and track progress of the operation using `Value()`. Alternatively, if you just want to wait until the operation completes, you can use `PollUntilDone()`.
+
+```C++
+  SomeServiceClient client;
+
+  auto operation = client.StartSomeLongRunningOperation();
+
+  while (!operation.IsDone())
+  {
+    Azure::Core::Http::RawResponse pollResponse = operation.Poll();

```

Comment:
why not use auto here to make the code less pedantic

Reply:
Also, I think it better to show PollUntilDone without the loop as this is the more common case and the code is much shorter.

Reply:
> why not use auto here to make the code less pedantic

I could do that, I was hoping to show more of the types/namespaces, but given `RawResponse` isn't something end users will care about right out the gate, I will use auto. Do you recommend I do that in a lot more places (auto everything) or just where it is not as relevant?

Reply:
I think showing that the code is simple is important; so yes, I think auto everywhere is good.